string data
---
string new_message